### ISSUE TITLE
DrawNode draws in wrong order

### ISSUE THREAD

**Issue by NgVThangBz** [axmolengine/axmol#1888]
- axmol version: All Version
- devices test on: All device
- developing environments: all platform
   - NDK version: r23c
   - Xcode version: 14.2+
   - Visual Studio: 
      - VS version: 2022 (17.9+)
      - MSVC version: 19.39+
      - Windows SDK version: 10.0.22621.0+
   - cmake version: 
Steps to Reproduce:

Looking at the code below, you can see that in the draw node the drawing order will be triangle, point, line:
![Screenshot 2024-05-07 at 14 51 17](https://github.com/axmolengine/axmol/assets/125545428/ed184d16-f7de-45d3-a9f4-55af1ca4bcfd)
So even if the triangle is added after the point, it is still drawn first, leading to the triangles always being covered by lines,

In my opinion: it would be good if lines, triangles, points are drawn one after another in the order they are added

can check code:
`DrawNode* _drawNode = DrawNode::create();`
`Director::getInstance()->getRunningScene()->addChild(_drawNode,100);`
 `_drawNode->drawLine(Vec2(200,200),Vec2(500,500),Color4B::RED);`
`_drawNode->drawSolidRect(Vec2(200,200),Vec2(500,500),Color4B::YELLOW);`

and result:

![Screenshot 2024-05-07 at 15 03 19](https://github.com/axmolengine/axmol/assets/125545428/13bc3432-b755-4e86-b4f4-9c101f2025d4)

**Comment by aismann** [comment]
Im working on a 'new' DrawNode class (currently available as extension DrawNodeEx).
I have seen this "wrong" behavior too.
This feature will be on a "next" version.

**Comment by aismann** [comment]
Workaround:
Create an own DrawNode object for each "drawing primitives".

**Comment by NgVThangBz** [comment]
> Workaround: Create an own DrawNode object for each "drawing primitives".

Yes, that's a temporary solution, or can convert line and point into polygons.

**Comment by aismann** [comment]
> > Workaround: Create an own DrawNode object for each "drawing primitives".
> 
> Yes, that's a temporary solution, or can convert line and point into polygons.

Thats not so easy as you think. There a lot of different scenarios for correct drawing order.
@halx99 
Please add tag:  'need founding' and anything like: 'workaround exist' and/or 'helpdesk' only

**Comment by aismann** [comment]
Add a method to set the correct draw order :
```
    drawNode->drawLine(Vec2(20, 20), Vec2(200, 200), Color4B::RED);
    drawNode->drawSolidRect(Vec2(50, 50), Vec2(150, 150), Color4B::YELLOW);
    drawNode->drawLine(Vec2(30, 20), Vec2(210, 200), Color4B::BLUE);
```
![image](https://github.com/axmolengine/axmol/assets/8652787/cd6bd32f-1d78-452b-b5e1-1bfa6bb3812e)

**Comment by aismann** [comment]
merged/solved with #1914
@NgVThangBz please close it.
